src/blob_store/recordfile.rs

```diff
@@ -0,0 +1,135 @@
+use byteorder::{BigEndian, ByteOrder};
+
+use std::cell::RefCell;
+use std::fs::{File, OpenOptions};
+use std::io;
+use std::io::{Read, Seek, SeekFrom, Write};
+use std::marker::PhantomData;
+use std::ops::Range;
+use std::path::Path;
+
+#[derive(Debug)]
+pub struct RecordFile<T> {
+    file: File,
+    current_offset: u64,
+    buf: RefCell<Vec<u8>>,
```

Comment:
Why is `buf` in the RecordFile struct and not just a stack variable in the calling functions if needed?